subliminal/providers/cinemast.py

```diff
@@ -0,0 +1,231 @@
+    languages = {Language.fromalpha2(l) for l in ['he']}
+    server_url = 'http://www.cinemast.org/he/cinemast/api/'
+    subtitle_class = CinemastSubtitle
+
+    default_username = 'subliminal@gmail.com'
+    default_password = 'subliminal'
+
+    def __init__(self, username=None, password=None):
+        if any((username, password)) and not all((username, password)):
+            raise ConfigurationError('Username and password must be specified')
+
+        self.session = None
+        self.username = username or self.default_username
+        self.password = password or self.default_password
+        self.user_id = None
+        self.token = None
+        self.session = None
+
+    def initialize(self):
+        self.session = Session()
+        self.session.headers['User-Agent'] = 'Subliminal/{}'.format(__short_version__)
+
+        # login
+        if self.username and self.password:
+            logger.debug('Logging in')
+            url = self.server_url + 'login/'
+
+            # actual login
+            data = {'username': self.username, 'password': self.password}
+            r = self.session.post(url, data=data, allow_redirects=False, timeout=10)
+
+            if r.status_code != 200:
+                raise AuthenticationError(self.username)
+
+            try:
+                result = r.json()
+                if 'token' not in result:
+                    raise AuthenticationError(self.username)
+
+                logger.info('Logged in')
+                self.user_id = r.json().get('user')
+                self.token = r.json().get('token')
+            except ValueError:
+                raise AuthenticationError(self.username)
+
+    def terminate(self):
+        # logout
+        if self.token or self.user_id:
+            logger.info('Logged out')
+            self.token = None
+            self.user_id = None
+
+        self.session.close()
+
+    def query(self, title, season=None, episode=None, year=None):
+        query = {
+            'q': title,
+            'user': self.user_id,
+            'token': self.token
+        }
+
+        # episode
+        if season and episode:
+            query['type'] = 'series'
+            query['season'] = season
+            query['episode'] = episode
+        else:
+            query['type'] = 'movies'
+            if year:
+                query['year_start'] = year - 1
+                query['year_end'] = year
+
+        # get the list of subtitles
+        logger.debug('Getting the list of subtitles')
+        url = self.server_url + 'search/'
+        r = self.session.post(url, data=query)
+        r.raise_for_status()
+
+        try:
+            results = r.json()
+        except ValueError:
+            return {}
+
+        # loop over results
+        subtitles = {}
+        for group_data in results.get('data', []):
+            # create page link
+            slug_name = group_data.get('name_en').lower().replace(' ', '-').replace('\'', '').replace('"', '')
```

Comment:
What about other characters? e.g. dot, semicolons
Is there any other way to get this information?

Reply:
Not really..
It's not my site so I'm just guessing and checking how to replace each character.
I'll added dot and semicolons though! Good idea.

Reply:
Make this a function so it doesn't look too long.